packages/gatsby-theme-carbon/src/components/Header/Header.module.scss

```diff
@@ -22,23 +11,70 @@ header.header {
   }
 }
 
+// Overrides upstream Carbon styles for header nav
+.headerWithHeaderNav :global(.bx--header__nav) {
+  @include carbon--breakpoint('lg') {
+    display: block;
+    position: absolute;
+    left: 25%;
+    top: 0;
+    padding-left: 0;
+  }
+}
+
+// Aligns HeaderNav with 12 col grid
+.headerWithHeaderNav :global(.bx--header__menu-bar) {
+  margin-left: 5px;
+}
+
+// Overrides upstream Carbon styles for header nav
+.headerWithHeaderNav :global(.bx--header__menu-item) {
+  font-size: 1rem;
+}
+
+// Overrides upstream Carbon styles for left nav in mobile.
 .header :global(.bx--header__action--menu) {
   border: none;
   top: 0.5px;
   position: relative;
   flex-shrink: 0;
-  &:focus {
-    outline: 2px solid $carbon--white-0;
-    outline-offset: -2px;
-  }
   @include carbon--breakpoint('lg') {
     display: none;
   }
 }
 
+// Overrides upstream Carbon styles for header nav
+.headerWithHeaderNav :global(.bx--header__nav::before) {
+  display: none;
+}
+
+.headerNameWithHeaderNav {
+  @include carbon--breakpoint('lg') {
+    padding: 0 2rem;
+  }
+}
+
+// Overrides upstream Carbon styles for left nav in mobile.
+.header :global(.bx--header__action--menu):focus {
+  outline: 2px solid $carbon--white-0;
+  outline-offset: -2px;
+}
+
+.header button {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+header.header {
```

Comment:
Did you need the extra specificity here? If so, just make a note of that so someone doesn't just assume they can remove it.